test/jdk/javax/swing/JToolTip/bug5047379.java

```diff
@@ -0,0 +1,170 @@
+import java.awt.Dimension;
+import java.awt.event.KeyEvent;
+import java.awt.GridLayout;
+import java.awt.Point;
+import java.awt.Robot;
+
+public class bug5047379 {
+    static JFrame frame;
+    static JButton a;
+    static JButton b;
+    static JButton c;
+    static JButton d;
+    static Robot testRobot;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        testRobot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                runTest();
+            }
+        });
+        testRobot.delay(1000);
+        testRobot.waitForIdle();
+        Point movePoint = getButtonPoint(b);
+        testRobot.mouseMove(movePoint.x, movePoint.y);
+        testRobot.delay(2000);
+        testRobot.waitForIdle();
+        handleToolTip();
+    }
+
+
+    static void handleToolTip() throws Exception {
+        SwingUtilities.updateComponentTreeUI(frame);
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    JToolTip tooltip = (JToolTip) Util.findSubComponent(
+                            JFrame.getFrames()[0], "JToolTip");
+
+                    MetalToolTipUI toolTipObj = (MetalToolTipUI) MetalToolTipUI.createUI(tooltip);
+
+                    if (tooltip == null) {
+                        throw new RuntimeException("Metal Tooltip not been found");
+                    }
+                    checkAcclString(toolTipObj, tooltip);
+
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                } finally {
+                    frame.dispose();
+                }
+            }
+        });
+    }
+
+    static void checkAcclString(MetalToolTipUI toolTipObj, JToolTip tooltip) {
+        toolTipObj.installUI(tooltip);
+
+        if (toolTipObj.getAcceleratorString() == null) {
+            throw new RuntimeException("Acceleration String for ToolTip of Metal L&F is null");
+        }
+    }
+
+    static Point getButtonPoint(JButton button) throws Exception {
+        final Point[] result = new Point[1];
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+
+            @Override
+            public void run() {
+                Point p = button.getLocationOnScreen();
+                Dimension size = button.getSize();
+                result[0] = new Point(p.x + size.width / 2, p.y + size.height / 2);
+            }
+        });
+        return result[0];
+    }
+
+    static void runTest() {
+        frame = new JFrame();
+        JTextArea area = new JTextArea();
+        JPanel p = new JPanel();
+
+        String text = "The mouse will hover over button B, so the\t\t\n";
+        text += "ToolTip will appear. Here is what should show\t\t\n";
+        text += "The word \\\"TEXT\\\" and then \\\"CTRL-B\\\"\\n\"\t\t";
+        text += "\n";
```

Comment:
Why are there two tabs in the end of each line.

Why can't you use static string concatenation? Or a text block?

Creating the string by `text += "literal"` isn't an efficient way.